sfge/src/main/java/com/salesforce/graph/ops/expander/ApexPathCollapserImpl.java

```diff
@@ -16,32 +16,51 @@
 final class ApexPathCollapserImpl implements ApexPathCollapser {
     private static final Logger LOGGER = LogManager.getLogger(ApexPathCollapserImpl.class);
 
+    private final PathExpansionRegistry registry;
+    private final Long pathExpansionId;
+
     /** This is the list of collapsers that the user has requested */
     private final List<ApexDynamicPathCollapser> dynamicPathCollapsers;
 
     /**
-     * Maps a ForkEvent to all of the ApexPathExpanders that have the same ForkEvent and returned a
-     * result. The ForkEvent passed to {@link #resultReturned(ApexPathExpander, ForkEvent,
-     * Optional)} are used to find all other ApexPathExpanders that might potentially be collapsed.
+     * Maps a ForkEvent Id to all of the ApexPathExpander Ids that have the same ForkEvent and
+     * returned a result. The ForkEvent passed to {@link #resultReturned(ApexPathExpander,
+     * ForkEvent, Optional)} are used to find all other ApexPathExpanders that might potentially be
+     * collapsed.
      */
-    private final Map<ForkEvent, List<ApexPathExpander>> forkEventToApexExpandersWithResults;
+    private final Map<Long, List<Long>> forkEventIdToApexExpanderIdsWithResults;
 
-    /** The list of all expanders that ApexPathExpanderUtil should remove on its next iteration */
-    private final List<ApexPathExpander> collapsedExpanders;
+    /**
+     * The list of Ids of all expanders that ApexPathExpanderUtil should remove on its next
+     * iteration
+     */
+    private final List<Long> collapsedApexPathExpanderIds;
```

Comment:
Logic remains the same as before, except we now deal with `ApexPathExpander` and `ForkEvent`'s Long ids instead of the actual instance.